src/status/wallet/collectibles.nim

```diff
@@ -0,0 +1,116 @@
+import strformat, httpclient, json, chronicles, sequtils, strutils, tables
+import ../libstatus/core as status
+import ../libstatus/contracts as contracts
+import eth/common/eth_types
+
+type Collectible* = ref object
+    name*, image*: string
+
+proc getTokenUri(contract: Contract, tokenId: int): string =
+  try:
+    let payload = %* [{
+      "to": $contract.address,
+      "data": contract.methods["tokenURI"].encodeAbi(tokenId)
+    }, "latest"]
+    let response = status.callPrivateRPC("eth_call", payload)
+    var postfixedResult: string = parseJson($response)["result"].str
```

Comment:
It might be a bit cleaner if we can use json deserialization for typing our json responses, example here:
https://github.com/status-im/nim-status-client/blob/365e5ab4c1871f656f9754db07ccb59c60763061/src/status/libstatus/accounts.nim#L26-L27